src/jsonrpccpp/server/connectors/httpserver.cpp

```diff
@@ -69,14 +119,17 @@ bool HttpServer::StartListening() {
             HttpServer::callback, this, MHD_OPTION_HTTPS_MEM_KEY,
             this->sslkey.c_str(), MHD_OPTION_HTTPS_MEM_CERT,
             this->sslcert.c_str(), MHD_OPTION_THREAD_POOL_SIZE, this->threads,
+            MHD_OPTION_SOCK_ADDR, addr.GetSockAddr(),
```

Comment:
I'm not sure if it is ok that you leak a stack pointer to libmicrohttpd here. The above proposed soluation (`bool SetAddrInfo(const std::string& addr, const int port)`) would also solve this. I couldn't find a statement in the libmicrohttpd documentation whether this is ok or not. But I guess it's not.

Reply:
That is a good point - I assumed that it is ok to pass a stack pointer and that `MHD_start_daemon` would make sure to extract all the data it needs without keeping references around.  That's why I put the `struct addrinfo` on the stack here with the RAII helper class.

But I agree that just putting it in the class does no harm and is safer as long as microhttpd does not clearly specify what life cycle the `struct sockaddr` should have.  I've changed that now accordingly.